### DISCUSSION TITLE
[@types/react-copy-to-clipboard] upgrade React 18

### DISCUSSION THREAD

**Discussion by getellez** [DefinitelyTyped/DefinitelyTyped#64209] [Issues with a @types package]
In React 18 + Typescript the component throws this error `'CopyToClipboard' cannot be used as a JSX component.`

[a comment by an automated account is not shown]

**Comment by nathanhannig** [comment]
they will never fix this, best to just make your own component from scratch

**Comment by FBerthelot** [comment]
If you are interested, I've proposed a fix on this PR : #69097

If you don't want to wait, you can add the .d.ts I've made in your own project and add something like that in your tsconfig.json :

```
{
"compilerOptions" : {
  "paths": {
        "react-copy-to-clipboard": ["./typings/react-copy-to-clipboard"]
   }
}
```

**Comment by t0yohei** [comment]
I have created a library called react-copy-to-clipboard-ts for React 18 and above.
This library follows react-copy-to-clipboard's functionality and type definitions are implemented in the library. 
Please use it if you like 😊

https://github.com/t0yohei/react-copy-to-clipboard-ts
